engine/src/test/java/io/zeebe/engine/processor/workflow/subprocess/EventSubprocessTest.java

```diff
@@ -106,12 +109,72 @@ public void shouldTriggerNestedEventSubproc() {
     assertEventSubprocessLifecycle(wfInstanceKey);
   }
 
+  @Test
+  public void shouldNotInterruptParentProcess() {
+    // given
+    final BpmnModelInstance model = createEventSubProc();
+    ENGINE.deployment().withXmlResource(model).deploy();
+
+    // when
+    final long wfInstanceKey =
+        ENGINE.workflowInstance().ofBpmnProcessId("proc").withVariable("innerKey", "123").create();
+    assertEventSubprocessLifecycle(wfInstanceKey);
+
+    ENGINE.job().ofInstance(wfInstanceKey).withType("type").complete().getKey();
+
+    // then
+    assertThat(
+            RecordingExporter.workflowInstanceRecords()
+                .withWorkflowInstanceKey(wfInstanceKey)
+                .limitToWorkflowInstanceCompleted())
+        .extracting(r -> tuple(r.getValue().getBpmnElementType(), r.getIntent()))
+        .containsSubsequence(
+            tuple(BpmnElementType.START_EVENT, WorkflowInstanceIntent.EVENT_OCCURRED),
+            tuple(BpmnElementType.SUB_PROCESS, WorkflowInstanceIntent.ELEMENT_ACTIVATED),
+            tuple(BpmnElementType.SUB_PROCESS, WorkflowInstanceIntent.ELEMENT_COMPLETED),
+            tuple(BpmnElementType.SERVICE_TASK, WorkflowInstanceIntent.ELEMENT_COMPLETED),
+            tuple(BpmnElementType.END_EVENT, WorkflowInstanceIntent.ELEMENT_COMPLETED),
+            tuple(BpmnElementType.PROCESS, WorkflowInstanceIntent.ELEMENT_COMPLETED));
+  }
+
+  @Test
+  public void shouldCompleteParentAfterEventSubprocCompletes() {
+    // given
+    final BpmnModelInstance model = createInterruptingEventSubproc();
+    ENGINE.deployment().withXmlResource(model).deploy();
+
+    // when
+    final long wfInstanceKey =
+        ENGINE.workflowInstance().ofBpmnProcessId("proc").withVariable("innerKey", "123").create();
+
+    TestUtil.waitUntil(
+        () ->
+            RecordingExporter.jobRecords(JobIntent.CREATED)
+                .withWorkflowInstanceKey(wfInstanceKey)
+                .exists());
```

Comment:
We don't need to use `TestUtil.waitUntil(..)`, the `RecordingExporter` waits until the job is created.

```
assertThat(
            RecordingExporter.jobRecords(JobIntent.CREATED)
                .withWorkflowInstanceKey(wfInstanceKey)
                .exists())
        .describedAs("Expected job to be created")
        .isTrue();
```

Reply:
I know but I thought it was more explicit to write "wait until job created exists". Besides, at that point, I'm not expecting it to fail, I just want to wait for it and I think this makes it more explicit. I guess it's a matter of preference

Reply:
Yes, correct. Since we are using the `RecordingExporter` in this way in many tests (i.e. because we never know when a record is processed), we should follow this pattern as long as it make sense :)